File: Assignment1/RightTriangle.java
/* A function that determines whether a triangle is a perpendicular one or not */

public class RightTriangle {
    public static void main(String[] args){
        int a = Integer.parseInt(args[0]);
        int b = Integer.parseInt(args[1]);
        int c = Integer.parseInt(args[2]);

        if (a > 0 & b > 0 & c > 0 & a * a + b * b == c * c){
            System.out.print("true");
        } else {
            System.out.print("false");
        }
    }
}

File: Assignment1/RightTriangle.c
#include <stdio.h>

/* RightTriangle that takes three int command-line arguments and determines whether they constitute the side lengths of some right triangle. */

int RightTriangle(int a, int b, int c){
  if (a > 0 & b > 0 & c > 0 & a * a + b * b == c * c){
    return 1;  /* true */
  } else {
    return 0;  /* false */
  }

}

int main(){
  int ra = RightTriangle(13, 12, 15);
  if(ra == 1){
    printf("true");
  } else {
    printf("false");
  }
}
